forc-pkg/src/pkg.rs

```diff
@@ -2055,7 +2077,47 @@ pub fn build(
     profile: &BuildProfile,
     outputs: &HashSet<NodeIx>,
     const_inject_map: &ConstInjectionMap,
-) -> anyhow::Result<Vec<(NodeIx, BuiltPackage)>> {
+) -> anyhow::Result<Vec<(NodeIx, BuildResult)>> {
+    #[allow(clippy::too_many_arguments)]
+    fn compile_util(
+        lib_namespace_map: &HashMap<NodeIx, namespace::Module>,
+        compiled_contract_deps: &HashMap<NodeIx, BuiltPackage>,
+        profile: &BuildProfile,
+        plan: &BuildPlan,
```

Comment:
Perhaps we can take the hint from clippy here and put some of these immutable arguments in a `CompileCtx` struct  or similar to cut down on the number of arguments? It could also include the `engines` and `target` too. Something like:

```rust
struct CompilePkgCtx<'a> {
    lib_namespace_map: &'a HashMap<NodeIx, namespace::Module>,
    compiled_contract_deps: &'a HashMap<NodeIx, BuiltPackage>,
    profile: &'a BuildProfile,
    plan: &'a BuildPlan,
    engines: Engines<'a>,
    target: &'a BuildTarget,
}
```